File: Tech Module/Software Technologies/20. Java Basics - Exercises/src/Phonebook_Upgrade.java
import java.util.*;

public class Phonebook_Upgrade {
    private static TreeMap<String, String> phonebook = new TreeMap<>();
    private static Scanner scanner = new Scanner(System.in);
    public static void main(String[] args) {

        while (TakeCommands()) {
        }
    }

    private static boolean TakeCommands() {
        String[] command = scanner.nextLine().split(" ");

        switch (command[0]) {
            case "A":
                String name = command[1];
                String number = command[2];
                add(name, number);
                break;
            case "S":
                name = command[1];
                search(name);
                break;
            case "ListAll":
                listAll();
                break;
            case "END":
                return false;
        }
        return true;
    }

    private static void listAll() {
        for(Map.Entry<String,String> contact: phonebook.entrySet()){
            System.out.println(contact.getKey() + " -> " + contact.getValue());
        }
    }

    private static void add(String name, String number) {
        phonebook.put(name, number);
    }

    private static void search(String name) {
        if (phonebook.containsKey(name)) {
            System.out.printf("%s -> %s%n", name, phonebook.get(name));
        } else {
            System.out.printf("Contact %s does not exist.%n", name);
        }
    }
}

File: 02. Tech Module/Programming Fundamentals/18. Dictionaries Lambda and LINQ - Exercises/Dictionaries Lambda and LINQ - Exercises/PhonebookUpgrade/PhonebookUpgrade.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PhonebookUpgrade
{
    class PhonebookUpgrade
    {
        static List<string> contacts = new List<string>();

        static Dictionary<string, string> phonebook = new Dictionary<string, string> { };

        static void AddToPhonebook(string name, string number)
        {
            if (phonebook.ContainsKey(name))
            {
                phonebook[name] = number;
            }
            else
            {
                phonebook.Add(name, number);
                contacts.Add(name);
            }
        }

        static void SearchInPhonebook(string name)
        {
            if (phonebook.ContainsKey(name))
            {
                Console.WriteLine($"{name} -> {phonebook[name]}");
            }
            else
            {
                Console.WriteLine($"Contact {name} does not exist.");
            }
        }

        static void ListAll()
        {
            contacts = contacts.OrderBy(x => x).ToList();
            PrintNames();
        }

        static void PrintNames()
        {
            foreach (string contact in contacts)
            {
                Console.WriteLine($"{contact} -> {phonebook[contact]}");
            }
        }

        static void Main(string[] args)
        {
            string command = "";

            while (command != "END")
            {
                string[] input = Console.ReadLine().Split(' ').ToArray();

                command = input[0];

                if (command == "A")
                {
                    string name = input[1];
                    string number = input[2];
                    AddToPhonebook(name, number);
                }
                else if (command == "S")
                {
                    string name = input[1];
                    SearchInPhonebook(name);
                }
                else if (command == "ListAll")
                {
                    ListAll();
                }
            }
        }
    }
}
